packages/slate-drop-or-paste-images/src/index.js

```diff
@@ -47,90 +46,80 @@ function DropOrPasteImages(options = {}) {
     return accepted
   }
 
-  /**
-   * Apply the change for a given file and update the editor with the result.
-   *
-   * @param {Change} change
-   * @param {Blob} file
-   * @return {Promise}
-   */
-
-  function asyncApplyChange(change, file) {
-    const { editor } = change
-
-    return Promise.resolve(insertImage(change, file)).then(() => {
-      editor.onChange(change)
-    })
-  }
-
   /**
    * On drop or paste.
    *
    * @param {Event} event
-   * @param {Change} change
+   * @param {Editor} editor
    * @param {Function} next
    * @return {State}
    */
 
-  function onInsert(event, change, next) {
-    const { editor } = change
+  function onInsert(event, editor, next) {
     const transfer = getEventTransfer(event)
     const range = getEventRange(event, editor)
 
-    switch (transfer.type) {
-      case 'files':
-        return onInsertFiles(event, change, next, transfer, range)
-      case 'html':
-        return onInsertHtml(event, change, next, transfer, range)
-      case 'text':
-        return onInsertText(event, change, next, transfer, range)
-      default:
-        return next()
+    if (
+      !acceptableTransferTypes ||
+      acceptableTransferTypes.includes(transfer.type)
+    ) {
+      switch (transfer.type) {
+        case 'files':
+          return onInsertFiles(event, editor, next, transfer, range)
+        case 'html':
+          return onInsertHtml(event, editor, next, transfer, range)
+        case 'text':
+          return onInsertText(event, editor, next, transfer, range)
+        default:
+          return next()
+      }
+    } else {
+      return next()
     }
   }
 
   /**
    * On drop or paste files.
    *
    * @param {Event} event
-   * @param {Change} change
+   * @param {Editor} editor
    * @param {Function} next
    * @param {Object} transfer
    * @param {Range} range
    * @return {Boolean}
    */
 
-  function onInsertFiles(event, change, next, transfer, range) {
+  function onInsertFiles(event, editor, next, transfer, range) {
     const { files } = transfer
 
     for (const file of files) {
```

Comment:
```
const filteredFiles = !extensions ? files : files.filter(file => {
  const type = file.type
  const [, etx] = type.split('/')
  return ext && matchExt(ext)
})
if (!filteredFiles.length) { return next() }
for (const file of filteredFiles) {
```

Reply:
I suggest moving the extension checking logic out of the loop. The reason is that this will enable the plugin to `return next()` if it determines that none of the extensions for the dropped files need to be inserted (IE a no op).